changelog.d/15808.misc

```diff
@@ -0,0 +1 @@
+Ignore key request if the device inbox is already big.
```

Comment:
Discussed in the triage meeting to remove key requests when we see cancellations as a no-risk first step here.

This may work but a bit worried about just dropping requests which would mean unable-to-decrypt errors for a client still expecting a response.

Reply:
Is this related to https://github.com/matrix-org/matrix-spec-proposals/pull/3944?

Reply:
Ohh thanks a lot @clokep, it will help me a lot, we didn't seem to know/remember that this MSC exists.

Following Keygan analysis and my experience with that trouble, I plan to replace this PR with one implementing at least the 2 first points of this MSC regarding `m.room_key_request`.